Marlin/Marlin_main.cpp

```diff
@@ -10257,14 +10290,23 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
       UNUSED(fr_mm_s);
       UNUSED(no_move);
 
-      // Set the new active extruder
-      active_extruder = tmp_extruder;
+      #if ENABLED(SWITCHING_EXTRUDER) && !DONT_SWITCH
+
+        stepper.synchronize();
+        move_extruder_servo(tmp_extruder);
+
+      #elif ENABLED(MK2_MULTIPLEXER)
+
+        if (tmp_extruder >= E_STEPPERS)
+          return invalid_extruder_error(tmp_extruder);
+
+        select_multiplexed_stepper(tmp_extruder);
```

Comment:
I've been looking at this some more, and I don't think `Planner` is the correct place to call `select_multiplexed_stepper`. I originally thought this because `Planner` currently enables any inactive extruder steppers.

`Planner::_buffer_line` places the commanded movements into a circular buffer. These commands are then processed in `Stepper::isr`. Enabling the extruders steppers in `Planner` works because it's not an issue to enable the steppers early (i.e. when they are queued as opposed to actually processed in `Stepper`). However, `select_multiplexed_stepper` can only be called immediately proceeding the first command destined for the newly selected stepper, otherwise step commands generated by `Stepper::isr` will be sent to wrong stepper while the buffer is processed to the point where the change was intended. `Stepper::isr` already knows which stepper the command being processed is destined for (via `current_block->active_extruder`, which is currently used for dual X carriage). As such, it should be straight forward to call `select_multiplexed_stepper` on a change to `current_block->active_extruder` in `Stepper::isr`. With this change, I don't believe it would be necessary to call `select_multiplexed_stepper` in any other location in Marlin.

We should probably also add a sanity check to ensure that `MIXING_EXTRUDER` and `MK2_MULTIPLEXER` are not both enabled.